### PR TITLE
Remove DYNARRAY and refactor for easier later C++ conversion

### PR DESCRIPTION
**Pull request by ni4**
This PR removes DYNARRAY and other #define-based array allocations, removing defs.h header.
Also it replaces direct field usage with accessor methods for easier later transition to C++ classes.

### CODE REVIEW

File: src/lib/rnp2.cpp [hunk 1]
```diff
@@ -3152,15 +3155,16 @@ key_get_uid_at(pgp_key_t *key, size_t idx, char **uid)
     if (!key || !uid) {
         return RNP_ERROR_NULL_POINTER;
     }
-    if (idx >= key->uidc || DYNARRAY_IS_EMPTY(key, uid)) {
+    if (idx >= pgp_get_userid_count(key)) {
```
Comment (randombit) [inline src/lib/rnp2.cpp]: This removes the check for UIDs being empty, does it matter?

Comment (ronaldtse) [reply src/lib/rnp2.cpp]: @ni4 could you have a look here? Thanks!

Comment (ni4) [reply src/lib/rnp2.cpp]: Empty uid list equals to `pgp_get_userid_count() == 0`, which will satisfy the `idx >=` check  since idx is unsigned.

**Comment by ni4** [comment]
@randombit Please see the last commit, I described all list elements there.

File: src/lib/rnp2.cpp (hunk 1 above)
Comment (dewyatt) [reply src/lib/rnp2.cpp]: Agree with @ni4, this should be good. It looks like the `DYNARRAY_IS_EMPTY` check here before was bogus (added by yours truly :P).

Comment (ronaldtse) [reply src/lib/rnp2.cpp]: Thanks guys!

**Pull request merged**